### ISSUE TITLE
NPE on widget removal

### ISSUE THREAD

**Issue by DeflatedPickle** [LWJGLX/lwjgl3-awt#30]
When removing an `AWTGLCanvas` from its container with `Container#remove(AWTGLCanvas)`, on Windows 10, I get the following NPE;
```
java.lang.NullPointerException
	at org.lwjgl.system.jawt.JAWTFunctions.JAWT_FreeDrawingSurface(JAWTFunctions.java:253)
	at org.lwjgl.opengl.awt.PlatformWin32GLCanvas.dispose(PlatformWin32GLCanvas.java:809)
	at org.lwjgl.opengl.awt.AWTGLCanvas.removeNotify(AWTGLCanvas.java:41)
	at java.awt.Container.removeNotify(Container.java:2807)
	at javax.swing.JComponent.removeNotify(JComponent.java:4758)
	at java.awt.Container.removeNotify(Container.java:2807)
	at javax.swing.JComponent.removeNotify(JComponent.java:4758)
	at java.awt.Container.remove(Container.java:1209)
	at java.awt.Container.remove(Container.java:1267)
```
My canvas is a still a skeleton, so I don't think I'm causing it;
```kotlin
class ModelViewerComponent : AWTGLCanvas() {
    override fun initGL() {
    }

    override fun paintGL() {
    }
}
```

**Comment by tisoft** [comment]
I have added a fix for the NPE in #28 

I would like to fix the root cause, though. Something is causing the Canvas to be disposed twice. Can you share your code? So that I could debug that?

**Comment by DeflatedPickle** [comment]
Thank you!

The code might be annoying to debug, as it uses generics and events, but the code causing it is [here](https://github.com/DeflatedPickle/Quiver/blob/18dcb8ba7019bcdd286924e0e3f94b71e2b188b2/filepanel/src/main/kotlin/com/deflatedpickle/quiver/filepanel/FilePanelPlugin.kt#L73-L77), with all the canvas stuff [here](https://github.com/DeflatedPickle/Quiver/blob/closes-%2322/modelviewer/src/main/kotlin/com/deflatedpickle/modelviewer/ModelViewerComponent.kt). The event is fired when a file is selected, causing the old file viewer (done by extension) to be removed and a new one to be added. The log shows what events are fired when, and it only shows one select file event per selection, so I'm not sure why it'd be happening twice, though it's possible the components list of the panel contains it twice?

Although, upon further inspection, I realise that it gets added to a `JScrollPane`. That *might* be it! I'll shift around some API and functionality to fix this without a special case and come back to close or comment if it works/doesn't work
